File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/AdminReturnDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.AdminLevel;
import com.example.oriengo.model.enumeration.Department;
import com.example.oriengo.model.enumeration.GenderType;
import lombok.*;
import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminReturnDTO {

    private Long id;

    private String firstName;

    private String lastName;

    private Integer age;

    private GenderType gender;

    private String phoneNumber;

    private String email;

    private Boolean enabled;

    private Boolean tokenExpired;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private LocalDateTime lastSeen;

    private Boolean suspended;

    private String suspensionReason;

    private LocalDateTime suspendedUntil;

    private LocalDateTime lastLoginAt;

    private LocalDateTime deletedAt;

    //private Set<String> roles;

    private AdminLevel adminLevel;

    private Long createdById;

    private Department department;

    private Boolean isDeleted;
}

File: OrienGO-Frontend/src/app/model/dto/AdminReturn.dto.ts
// admin-return.dto.ts

import { AdminLevel } from "../enum/AdminLevel.dto";
import { Department } from "../enum/Department.enum";
import { GenderType } from "../enum/GenderType.enum";

export interface AdminReturnDTO {
  id: number;
  firstName: string;
  lastName: string;
  age: number;
  gender: GenderType;
  phoneNumber?: string;
  email: string;
  enabled: boolean;
  tokenExpired: boolean;
  createdAt?: string;
  updatedAt?: string;
  lastSeen?: string;
  suspended: boolean;
  suspensionReason?: string;
  suspendedUntil?: string;
  lastLoginAt?: string;
  deletedAt?: string;
  //roles: string[];
  adminLevel: AdminLevel;
  createdById: number;
  department: Department;
  isDeleted: boolean;
}
